internal_ws/ykcompile/src/arch/x86_64/mod.rs

```diff
@@ -1378,28 +1376,26 @@ impl TraceCompiler {
                 }
             }
 
-            // Create a vector of `FrameInfo`s in which we store the allocated memory pointers,
-            // body, and basic block index for each active stack frame during the guard failure.
-            // This vector is later handed to the StopgapInterpreter, in order to initialise the
-            // stack frames needed to interpret SIR. We can use a fixed register here, since we've
-            // spilled all registers to the stack.
+            // The trace inlines several functions, so we now have to split the "sort of single
+            // frame" trace into a multi-frame vector so that the StopgapInterpreter can execute
+            // normal SIR. Fortunately all the variables we care about are on the stack at this
+            // point which simplifies this process.
```

Comment:
If I was being super nitpicky, I would like to specify why this simplifies the process, i.e. because we can use registers freely without worrying about clobbering stuff. But I'm not going to insist on it.

Reply:
Maybe deleting "which simplifies this process" is the easy fix?

Reply:
I've done a slightly bigger fix than I first intended, but I hope it addresses this -- and more fully than I first intended! https://github.com/softdevteam/yk/pull/285/commits/bb3a1fda28b0acd72d7150c4fe53f1a3717c5dec